crates/edr_napi/src/trace/solidity_stack_trace.rs

```diff
@@ -0,0 +1,398 @@
+    pub type_: StackTraceEntryTypeConst<{ StackTraceEntryType::CALL_FAILED_ERROR as u8 }>,
+    pub source_reference: SourceReference,
+}
+#[napi(object)]
+pub struct DirectLibraryCallErrorStackTraceEntry {
+    #[napi(
+        js_name = "type",
+        ts_type = "StackTraceEntryType.DIRECT_LIBRARY_CALL_ERROR"
+    )]
+    pub type_: StackTraceEntryTypeConst<{ StackTraceEntryType::DIRECT_LIBRARY_CALL_ERROR as u8 }>,
+    pub source_reference: SourceReference,
+}
+
+#[napi(object)]
+pub struct UnrecognizedCreateErrorStackTraceEntry {
+    #[napi(
+        js_name = "type",
+        ts_type = "StackTraceEntryType.UNRECOGNIZED_CREATE_ERROR"
+    )]
+    pub type_: StackTraceEntryTypeConst<{ StackTraceEntryType::UNRECOGNIZED_CREATE_ERROR as u8 }>,
+    pub message: ClassInstance<ReturnData>,
+    pub source_reference: Option<Undefined>,
+    pub is_invalid_opcode_error: bool,
+}
+
+#[napi(object)]
+pub struct UnrecognizedContractErrorStackTraceEntry {
+    #[napi(
+        js_name = "type",
+        ts_type = "StackTraceEntryType.UNRECOGNIZED_CONTRACT_ERROR"
+    )]
+    pub type_: StackTraceEntryTypeConst<{ StackTraceEntryType::UNRECOGNIZED_CONTRACT_ERROR as u8 }>,
+    pub address: Uint8Array,
+    pub message: ClassInstance<ReturnData>,
+    pub source_reference: Option<Undefined>,
+    pub is_invalid_opcode_error: bool,
+}
+
+#[napi(object)]
+pub struct OtherExecutionErrorStackTraceEntry {
+    #[napi(
+        js_name = "type",
+        ts_type = "StackTraceEntryType.OTHER_EXECUTION_ERROR"
+    )]
+    pub type_: StackTraceEntryTypeConst<{ StackTraceEntryType::OTHER_EXECUTION_ERROR as u8 }>,
+    pub source_reference: Option<SourceReference>,
+}
+
+#[napi(object)]
+pub struct ContractTooLargeErrorStackTraceEntry {
+    #[napi(
+        js_name = "type",
+        ts_type = "StackTraceEntryType.CONTRACT_TOO_LARGE_ERROR"
+    )]
+    pub type_: StackTraceEntryTypeConst<{ StackTraceEntryType::CONTRACT_TOO_LARGE_ERROR as u8 }>,
+    pub source_reference: Option<SourceReference>,
+}
+
+#[napi(object)]
+pub struct InternalFunctionCallStackEntry {
+    #[napi(
+        js_name = "type",
+        ts_type = "StackTraceEntryType.INTERNAL_FUNCTION_CALLSTACK_ENTRY"
+    )]
+    pub type_:
+        StackTraceEntryTypeConst<{ StackTraceEntryType::INTERNAL_FUNCTION_CALLSTACK_ENTRY as u8 }>,
+    pub pc: u32,
+    pub source_reference: SourceReference,
+}
+
+#[napi(object)]
+pub struct ContractCallRunOutOfGasError {
+    #[napi(
+        js_name = "type",
+        ts_type = "StackTraceEntryType.CONTRACT_CALL_RUN_OUT_OF_GAS_ERROR"
+    )]
+    pub type_:
+        StackTraceEntryTypeConst<{ StackTraceEntryType::CONTRACT_CALL_RUN_OUT_OF_GAS_ERROR as u8 }>,
+    pub source_reference: Option<SourceReference>,
+}
+
+#[allow(dead_code)]
+// NOTE: This ported directly from JS for completeness, however the type must be
+// used verbatim in JS definitions because napi-rs does not store not allows to
+// reuse the same type unless fully specified at definition site.
```

Comment:
Typo? The comment isn't clear to me

Reply:
Yeah, I could have done better job at explaining this. I tried to be more in-depth about the context for this: the gist is that the type alias is mostly for the Rust side of the bindings. To actually type-check we would have to not use the type alias but the entire type as is, which is very bad, considering it takes 26 lines, so we accept the trade-off and use the type alias (which type-checks to `any` atm).

When we will limit the exposed API interface, we might specify the type literally in few select places if we deem the trade-off in readability to be acceptable :ok_hand:

Hope that makes sense and it's okay.